ethcore/src/miner/miner.rs

```diff
@@ -291,31 +291,13 @@ impl Miner {
 	fn map_existing_pending_block<F, T>(&self, f: F, latest_block_number: BlockNumber) -> Option<T> where
 		F: FnOnce(&ClosedBlock) -> T,
 	{
-		self.from_pending_block(
-			latest_block_number,
-			|| None,
-			|block| Some(f(block)),
-		)
-	}
-
-	// TODO [ToDr] Get rid of this method.
-	//
-	// We should never fall back to client, this can be handled on RPC level by returning Option<>
-	fn from_pending_block<H, F, G>(&self, latest_block_number: BlockNumber, from_chain: F, map_block: G) -> H where
-		F: Fn() -> H,
-		G: FnOnce(&ClosedBlock) -> H,
-	{
-		let sealing = self.sealing.lock();
-		sealing.queue.peek_last_ref().map_or_else(
-			|| from_chain(),
-			|b| {
-				if b.block().header().number() > latest_block_number {
-					map_block(b)
-				} else {
-					from_chain()
-				}
-			}
-		)
+		self.sealing.lock().queue
+			.peek_last_ref()
+			.and_then(|b| if b.block().header().number() > latest_block_number {
+				Some(f(b))
+			} else {
+				None
+			})
```

Comment:
Just to note, at some point in future this would be possible to write using `Option::filter` that's now [implemented, but unstable](https://github.com/rust-lang/rust/issues/45860).

Reply:
Should these kind of notes be added as "to-do" comments within the code?

Reply:
It's not _that_ important IMO, but maybe we should.